list.h

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+/* Defines */
+
+/* Starting capacity for array lists. */
+#define LIST_STARTING_CAPACITY 32
+
+/* Error code if an item does not exists in the array list. */
+#define ITEM_NOT_FOUND -1
+
+/* Macro that checks if bytes are equal a to value b. */
+#define BYTES_EQUAL(a, b, size, pred)   \
+    do {                                \
+        size_t __size = (size);         \
+        char *__a = (a);                \
+        char *__b = (b);                \
+        do {                            \
+            pred = *__a++ == *__b++;    \
+        } while(pred && --__size > 0);  \
+    } while(0)                          
```

Comment:
There is a builtin function in <string.h> called memcmp() that already does this. This macro is fine though, we can keep it. sizeof(char) is not guaranteed to be 1 byte! It may be more or less depending on the architecture. Since this assumed we are dealing with single bytes, we may use uint8_t from the <stdint.h> library. We could also create our own aliases for the different ints to make the code less verbose.

#define u8 uint8_t
#define i8 int8_t
#define u16 uint16_t
#define i16 int16_t

etc.

Reply:
I must say I do agree with the latter statement, but I believe using the string library to be adding a lot of unused bloat.